### PR TITLE
[main] Update framework to version 10

### PR DESCRIPTION
**Pull request by Kipjr**

### CODE REVIEW

File: .github/workflows /docker-publish.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  # See also https://docs.docker.com/docker-hub/builds/automated-testing/
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Run tests
+        run: |
+          if [ -f docker-compose.test.yml ]; then
```
Comment (SubSide) [inline .github/workflows /docker-publish.yml:27]: What does this if do?

File: .github/workflows /docker-publish.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        uses: actions/checkout@v2
+
+      # Login against a Docker registry except on PR
+      # https://github.com/docker/login-action
+      - name: Log into registry ${{ env.REGISTRY }}
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@28218f9b04b4f3f62068d7b6ce6ca5b26e35336c
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
```
Comment (SubSide) [inline .github/workflows /docker-publish.yml:55]: Does this work? Feels weird that username/password still works when the actor changes

File: .github/workflows /docker-publish.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      # Login against a Docker registry except on PR
+      # https://github.com/docker/login-action
+      - name: Log into registry ${{ env.REGISTRY }}
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@28218f9b04b4f3f62068d7b6ce6ca5b26e35336c
```
Comment (SubSide) [inline .github/workflows /docker-publish.yml:52]: Why is this scoped to such a weird version? Can't it just be @v2 or something?

Comment (SubSide) [reply .github/workflows /docker-publish.yml:52]: Same applies to the other `uses` sections

File: .github/workflows /docker-publish.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
+name: Docker
```
Comment (SubSide) [inline .github/workflows /docker-publish.yml:1]: Does the folder `workflows` contain a space? xd

File: .github/workflows /laravel.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+jobs:
+  laravel-tests:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Copy .env
+        run: php -r "file_exists('.env') || copy('.env.example', '.env');"
```
Comment (SubSide) [inline .github/workflows /laravel.yml:17]: I love this :)

File: app/Exceptions/Handler.php [hunk 6]
```diff
@@ -2,75 +2,29 @@
...
-    /**
-     * Render an exception into an HTTP response.
-     *
-     * @param  \Illuminate\Http\Request  $request
-     * @param  \Exception  $exception
-     * @return \Illuminate\Http\Response
-     */
-    public function render($request, Exception $exception)
-    {
-        if($this->isHttpException($exception))
```
Comment (SubSide) [inline app/Exceptions/Handler.php:52]: Where did this code go?

File: app/Exceptions/Handler.php [hunk 7]
```diff
@@ -2,75 +2,29 @@
...
-                default:
-                    return $this->renderHttpException($exception);
-                    break;
-            }
-        }
-        else
-        {
-            return parent::render($request, $exception);
-        }
+        $this->reportable(function (Throwable $e) {
```
Comment (SubSide) [inline app/Exceptions/Handler.php:26]: What does this do?

File: app/Models/User.php [hunk 8]
```diff
@@ -2,31 +2,47 @@
...
+use Illuminate\Database\Eloquent\Factories\HasFactory;
 use Illuminate\Foundation\Auth\User as Authenticatable;
+use Illuminate\Notifications\Notifiable;
+use Laravel\Sanctum\HasApiTokens;
 
 class User extends Authenticatable
 {
-    use Notifiable;
-
+    use HasApiTokens, HasFactory, Notifiable;
```
Comment (SubSide) [inline app/Models/User.php:13]: Not sure if these two should be added

File: routes/web.php [hunk 9]
```diff
@@ -1,53 +1,67 @@
...
+Route::get('faq/list', [FaqController::class, 'listFaqs'])->name('faq.list');
+Route::match(['get', 'post'], 'faq/add',  [FaqController::class, 'edit'])->name('faq.add');
+Route::match(['get', 'post'], 'faq/edit/{id}',  [FaqController::class, 'edit'])->name('faq.edit');
+Route::match(['get', 'post'], 'faq/delete/{id}',  [FaqController::class, 'delete'])->name('faq.delete');
 
 // Login and log out
-Route::get('login/{token}', 'LoginController@loginByToken');
-Route::match(['get', 'post'], 'login', 'LoginController@login')->name('login');
-Route::post('logout', 'LoginController@logout')->name('logout');
+Route::get('login/{token}',  [LoginController::class, 'loginByToken']);
```
Comment (Raeolu) [inline routes/web.php:27]: Maybe this route could also have a name for consistencyy

File: app/Providers/AppServiceProvider.php [hunk 10]
```diff
@@ -13,9 +13,11 @@ class AppServiceProvider extends ServiceProvider
      *
      * @return void
      */
-    public function boot()
+    public function boot(): void
     {
         Schema::defaultStringLength(191);
+        \URL::forceScheme('https');
```
Comment (SubSide) [inline app/Providers/AppServiceProvider.php:19]: Maybe this needs to be added with an if to see if you're in debug mode. This could possibly break development, and otherwise maybe the debug bar

File: app/Providers/AuthServiceProvider.php [hunk 11]
```diff
@@ -2,15 +2,15 @@
 
 namespace App\Providers;
 
-use Illuminate\Support\Facades\Gate;
+// use Illuminate\Support\Facades\Gate;
```
Comment (SubSide) [inline app/Providers/AuthServiceProvider.php:5]: Commented

File: app/Providers/EventServiceProvider.php [hunk 12]
```diff
@@ -25,14 +25,23 @@ class EventServiceProvider extends ServiceProvider
     ];
 
     /**
-     * Register any events for your application.
+     * The event to listener mappings for the application.
      *
-     * @return void
+     * @var array<class-string, array<int, class-string>>
```
Comment (SubSide) [inline app/Providers/EventServiceProvider.php:30]: What does this @ var do?

File: app/Providers/RouteServiceProvider.php [hunk 13]
```diff
@@ -2,72 +2,39 @@
...
-
-        parent::boot();
-    }
-
-    /**
-     * Define the routes for the application.
-     *
-     * @return void
-     */
-    public function map()
```
Comment (SubSide) [inline app/Providers/RouteServiceProvider.php:36]: Where does this happen now? Or is this not needed anymore?

Comment (SubSide) [reply app/Providers/RouteServiceProvider.php:36]: N.v.m. I see it a couple lines below haha :blush:

File: config/auth.php [hunk 14]
```diff
@@ -94,7 +94,7 @@
     'passwords' => [
         'users' => [
             'provider' => 'users',
-            'table' => 'password_resets',
+            'table' => 'password_reset_tokens',
```
Comment (SubSide) [inline config/auth.php:97]: Is this change necessary? Why can't the old table be used?

File: database/factories/UserFactory.php [hunk 15]
```diff
@@ -1,25 +1,38 @@
...
+     *
+     * @return array<string, mixed>
+     */
+    public function definition(): array
+    {
+        return [
+            'name' => fake()->name(),
+            'email' => fake()->unique()->safeEmail(),
+            'email_verified_at' => now(),
+            'password' => '$2y$10$92IXUNpkjO0rOQ5byMi.Ye4oKoEa3Ro9llC/.og/at2.uheWG/igi', // password
```
Comment (SubSide) [inline database/factories/UserFactory.php:24]: why not use bcrypt like before?

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 16]
```diff
@@ -4,14 +4,14 @@
 use Illuminate\Database\Schema\Blueprint;
 use Illuminate\Database\Migrations\Migration;
 
-class CreateUsersTable extends Migration
+return new class extends Migration
```
Comment (SubSide) [inline database/migrations/2014_10_12_000000_create_users_table.php:7]: Is `return new class` correct here? This seems wrong.

File: public/index.php [hunk 17]
```diff
@@ -1,60 +1,55 @@
...
-| will load up this application so that we can run it and send
-| the responses back to the browser and delight our users.
+| Composer provides a convenient, automatically generated class loader for
+| this application. We just need to utilize it! We'll simply require it
+| into the script here so we don't need to manually load our classes.
 |
 */
 
-$app = require_once __DIR__.'/../bootstrap/app.php';
+require __DIR__.'/../vendor/autoload.php';
```
Comment (SubSide) [inline public/index.php:34]: I really hope you know what you're doing in this file 😅

File: routes/api.php [hunk 18]
```diff
@@ -13,6 +13,6 @@
 |
 */
 
-Route::middleware('auth:api')->get('/user', function (Request $request) {
+Route::middleware('auth:sanctum')->get('/user', function (Request $request) {
```
Comment (SubSide) [inline routes/api.php:16]: Is this correct? Why is it changed from api to sanctum?

File: routes/web.php [hunk 19]
```diff
@@ -1,53 +1,67 @@
...
+| Web Routes
+|--------------------------------------------------------------------------
+|
+| Here is where you can register web routes for your application. These
+| routes are loaded by the RouteServiceProvider and all of them will
+| be assigned to the "web" middleware group. Make something great!
+|
+*/
+
+Route::get('/', [HomeController::class, 'index'])->name('home');
```
Comment (SubSide) [inline routes/web.php:17]: Here too I hope you know what you're doing haha. Hopefully all went well

**Review by SubSide — CHANGES_REQUESTED** [review]
See comments

File: .github/workflows /docker-publish.yml (hunk 1 above)
Comment (Kipjr) [reply .github/workflows /docker-publish.yml:27]: If docker-compose.test.yml exist then it will use the contents of that specific file, else it will use default.

File: .github/workflows /docker-publish.yml (hunk 3 above)
Comment (Kipjr) [reply .github/workflows /docker-publish.yml:52]: > A digest is an id that is automatically created during build time and cannot be changed (immutable). When an image is pulled using a digest, a docker pull will download the same image every time on any os/arch. This is called image pinning.

Security.. I didn't create it...

File: .github/workflows /docker-publish.yml (hunk 2 above)
Comment (Kipjr) [reply .github/workflows /docker-publish.yml:55]: yes. See https://github.com/kipjr/docker-piwigo

File: config/cors.php [hunk 20]
```diff
@@ -0,0 +1,34 @@
...
+
+    'allowed_origins_patterns' => [],
+
+    'allowed_headers' => ['*'],
+
+    'exposed_headers' => [],
+
+    'max_age' => 0,
+
+    'supports_credentials' => false,
```
Comment (Raeolu) [inline config/cors.php:32]: Maybe this should be set to true at some point in the future if authentication is necessary for the api/* routes?

File: app/Exceptions/Handler.php (hunk 7 above)
Comment (Kipjr) [inline app/Exceptions/Handler.php:26]: This was updated from Laravel itself. See commit here: 
https://github.com/laravel/laravel/commit/4931af14006610bf8fd1f860cea1117c68133e94

File: app/Models/User.php (hunk 8 above)
Comment (Kipjr) [reply app/Models/User.php:13]: this was added upstream in version 8.x : https://github.com/laravel/laravel/pull/5663

**Review by Raeolu — COMMENTED** [review]
Welp I don't think I added much with this review but I hope it's still a bit useful :3

File: app/Providers/AppServiceProvider.php (hunk 10 above)
Comment (Kipjr) [reply app/Providers/AppServiceProvider.php:19]: Serving without this in a remote development or production causes issues. The only usecase would be http://localhost 
Also see https://learn.microsoft.com/en-us/azure/active-directory/develop/reply-url#supported-schemes

File: app/Providers/EventServiceProvider.php (hunk 12 above)
Comment (Kipjr) [reply app/Providers/EventServiceProvider.php:30]: Also upstream. I think this silences the output.

File: routes/web.php (hunk 9 above)
Comment (SubSide) [reply routes/web.php:27]: The reason I didn't add a name to this one in the past, is because with a name you allow to redirect/link to it through code. So in a way I tried to enforce here that you can only do it by manually putting in the URL in the browser

File: .github/workflows /docker-publish.yml (hunk 2 above)
Comment (SubSide) [reply .github/workflows /docker-publish.yml]: Have you tested it with someone else's account?

File: .github/workflows /docker-publish.yml (hunk 3 above)
Comment (SubSide) [reply .github/workflows /docker-publish.yml]: That doesn't make sense to me, what has security to do with this?

File: app/Models/User.php (hunk 8 above)
Comment (SubSide) [reply app/Models/User.php:13]: Ok, but is it needed for us?

File: app/Providers/AppServiceProvider.php (hunk 10 above)
Comment (SubSide) [reply app/Providers/AppServiceProvider.php:19]: Well, production isn't an issue then, because that wouldn't be in debug mode :P
If this is really an issue in a remote environment we should add a second environment variable to handle that case.

File: app/Providers/EventServiceProvider.php (hunk 12 above)
Comment (Raeolu) [reply app/Providers/EventServiceProvider.php:30]: I feel like something got a bit mixed up here. Some example EventServiceProvider stuff I found online looked like this:
`class EventServiceProvider extends ServiceProvider
{
    /**
     * The event to listener mappings for the application.
     *
     * @var array<class-string, array<int, class-string>>
     */
    protected $listen = [
        Registered::class => [
            SendEmailVerificationNotification::class,
        ],
    ];

    /**
     * Register any events for your application.
     */
    public function boot(): void
    {
        Ad::observe(AdObserver::class);
    }

    /**
     * Determine if events and listeners should be automatically discovered.
     */
    public function shouldDiscoverEvents(): bool
    {
        return false;
    }
}`

So I think either the $listen property should be added + the boot() comment should be re-added, or: leave out the $listen and just correct the comment to actually describe the boot() method.
